proposals/continuations/examples/actor-lwt.wast

```diff
@@ -0,0 +1,355 @@
+;; Actors via lightweight threads
+
+;; actor interface
+(module $actor
+  (type $func (func))
+  (type $cont (cont $func))
+
+  (event $self (export "self") (result i32))
+  (event $spawn (export "spawn") (param (ref $cont)) (result i32))
+  (event $send (export "send") (param i32 i32))
+  (event $recv (export "recv") (result i32))
+)
+(register "actor")
+
+;; a simple example - pass a message through a chain of actors
+(module $chain
+  (type $func (func))
+  (type $cont (cont $func))
+
+  (type $i-func (func (param i32)))
+  (type $i-cont (cont $i-func))
+
+  (event $self (import "actor" "self") (result i32))
+  (event $spawn (import "actor" "spawn") (param (ref $cont)) (result i32))
+  (event $send (import "actor" "send") (param i32 i32))
+  (event $recv (import "actor" "recv") (result i32))
+
+  (elem declare func $next)
+
+  (func $log (import "spectest" "print_i32") (param i32))
+
+  (func $next (param $p i32)
+    (local $s i32)
+    (local.set $s (suspend $recv))
+    (call $log (i32.const -1))
+    (suspend $send (local.get $s) (local.get $p))
+  )
+
+  (func $spawnMany (param $p i32) (param $n i32)
+    (if (i32.eqz (local.get $n))
+      (then (suspend $send (i32.const 42) (local.get $p))
+            (return))
+      (else (return_call $spawnMany (suspend $spawn (cont.bind (type $cont) (local.get $p) (cont.new (type $i-cont) (ref.func $next))))
```

Comment:
Does this need tail calls or would a loop work just as well? (For Wasm, I'd use loops where possible.)

Reply:
It's perfectly possible to do with a loop. It isn't a loop currently because I transcribed it from functional code I had written elsewhere. I'll change it.